include/seahorn/Expr/ExprOpCore.hh

```diff
@@ -46,8 +53,8 @@ struct TerminalBase : public expr::Operator {
 
 /// \brief Terminal Operator
 /// \p T is the type wrapped by the terminal, and \p P is the implementation of
-/// terminal traits for \p T
-template <typename T, typename P = TerminalTrait<T>>
+/// terminal traits for \p T , C is the tyepChecker
+template <typename T, typename P = TerminalTrait<T>, typename C = typeCheck::Any>
```

Comment:
I think type checking should be part of `TerminalTrait`, not sometehing specified on the side. Just like printing and ordering.